x-pack/platform/plugins/shared/index_management/server/routes/api/component_templates/schema_validation.ts

```diff
@@ -19,6 +19,7 @@ export const componentTemplateSchema = schema.object({
         data_retention: schema.maybe(schema.string()),
       })
     ),
+    data_stream_options: schema.maybe(schema.object({}, { unknowns: 'allow' })),
```

Comment:
nit: can we add a small comment here on why we are allowing unknowns for the time being?